src/librawspeed/decoders/Cr2Decoder.cpp

```diff
@@ -314,10 +314,10 @@ void Cr2Decoder::sRawInterpolate() {
 
 // Note: Thread safe.
 
-template <typename T1, typename T2>
-void _interpolate_422(T1&& yuv2rgb, T2&& store, const int* sraw_coeffs,
-                      RawImage& mRaw, int hue, int hue_last, int w, int h,
-                      int start_h, int end_h) {
+template <typename T>
+static inline void interpolate_422(T yuv2rgb, const int* sraw_coeffs,
```

Comment:
Minor cosmetic comment: the `inline` here does not make a lot of sense. I don't think the compiler would inline this huge function and even if, it would not make any measurable difference, since it is only one function call per complete image.